File: app/src/main/java/ru/p3tr0vich/fuel/BroadcastReceiverLocalBase.java
package ru.p3tr0vich.fuel;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.IntentFilter;
import android.support.annotation.NonNull;
import android.support.v4.content.LocalBroadcastManager;

public abstract class BroadcastReceiverLocalBase extends BroadcastReceiver {

    protected abstract String getAction();

    public final void register(@NonNull Context context) {
        LocalBroadcastManager.getInstance(context).registerReceiver(this, new IntentFilter(getAction()));
    }

    public final void unregister(@NonNull Context context) {
        LocalBroadcastManager.getInstance(context).unregisterReceiver(this);
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/receivers/BroadcastReceiverLocalBase.kt
package ru.p3tr0vich.fuel.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import androidx.localbroadcastmanager.content.LocalBroadcastManager

abstract class BroadcastReceiverLocalBase : BroadcastReceiver() {
    protected abstract val action: String

    fun register(context: Context) {
        LocalBroadcastManager.getInstance(context).registerReceiver(this, IntentFilter(action))
    }

    fun unregister(context: Context) {
        LocalBroadcastManager.getInstance(context).unregisterReceiver(this)
    }

    companion object {
        @JvmStatic
        fun send(context: Context, intent: Intent) {
            LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
        }
    }
}